lib/Elastica/ResultSet.php

```diff
@@ -15,117 +15,45 @@
 class ResultSet implements \Iterator, \Countable, \ArrayAccess
 {
     /**
-     * Class for the static create method to use.
-     *
-     * @var string
-     */
-    protected static $_class = 'Elastica\\ResultSet';
-
-    /**
      * Results.
      *
-     * @var array Results
+     * @var Result[] Results
      */
-    protected $_results = array();
+    private $_results = array();
```

Comment:
In the last discussion I explained that this is a good change and we should add it to CHANGELOG and bump the major version - manipulating the ResultSet by inheritance isnt correct anymore - changing the implementation of the ResultSetBuilder is the way to manipulate it now.